api/services/user_contact_service.py

```diff
@@ -86,10 +96,27 @@ def process_custom_fields(self, jsonData, user_custom_field_data=False):
             ):
                 userdata = self.get_user_custom_fields_object(field_name, field_value)
                 user_custom_contact_data.extend(userdata)
-            elif custom_fields_conditions:
+            elif custom_fields_conditions and not self.check_if_exist(
+                active_custom_fields_key_values, field_name, field_value
+            ):
                 models.UserCustomFields.query.set_custom_field_as_active(
                     field_name, field_value, self.user_phone
                 )
+            elif custom_fields_conditions and self.check_if_exist(
+                active_custom_fields_key_values, field_name, field_value
+            ):
+                active_custom_fields_key_values.pop(
+                    str(field_name) + "_" + str(field_value)
+                )
```

Comment:
Removing the custom fields that are present in active fields. We are removing this as in the last we want only those custom fields that are active in our database but are inactive in the RapidPro